lib/matplotlib/widgets.py

```diff
@@ -435,6 +446,15 @@ def _update(self, event):
         if val not in [None, self.val]:
             self.set_val(val)
 
+    def _format(self, val):
+        """Pretty-print *val*."""
+        if self.valfmt is not None:
+            return self.valfmt % val
+        else:
+            _, s, _ = self._fmt.format_ticks([self.valmin, val, self.valmax])
+            # fmt.get_offset is actually the multiplicative factor, if any.
+            return s + self._fmt.get_offset()
```

Comment:
This will result in 3e-8 to be shown as 31e-8.

Reply:
oops, I had `axes.formatter.usemathtext` in my rcdefaults, we just need to force that here...